traits/tests/test_get_traits.py

```diff
@@ -0,0 +1,62 @@
+#------------------------------------------------------------------------------
+#
+#  Copyright (c) 2007, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in /LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+#------------------------------------------------------------------------------
+
+from __future__ import absolute_import
+
+from traits.testing.unittest_tools import unittest
+
+from ..api import HasTraits, Int, Str, Undefined, ReadOnly, Float
+
+
+class Foo(HasTraits):
+    num = Int
+    bar = Str
+    
+
+class GetTraitTestCase(unittest.TestCase):
```

Comment:
The tests fail without the changed functionality because the class trait definition was overruling the dynamically defined trait. 
I will be adding a test for trait_view()